sentry_sdk/metrics.py

```diff
@@ -449,7 +462,16 @@ def _emit(
         encoded_metrics = _encode_metrics(flushable_buckets)
         metric_item = Item(payload=encoded_metrics, type="statsd")
         envelope = Envelope(items=[metric_item])
-        self._capture_func(envelope)
+
+        # A malfunctioning transport might create a forever loop of metric
+        # emission when it emits a metric in capture_envelope.  We still
+        # allow the capture to take place, but interior metric incr calls
+        # or similar will be disabled.  In the background thread this can
+        # never happen, but in the force flush case which happens in the
+        # foreground we might make it here unprotected.
+        with recursion_protection():
```

Comment:
In the background thread, this is basically always a no-op as `in_metrics` is forcefully set to `True`. In the force flush case we might be flipping it here. Why is `flush` itself unprotected? Two reasons:

1. on the sentry side we currently patch it, so making it a noop would break the logic there
2. it's potentially brittle since `_emit` is invoked in more than one place, this should be safer.